docs/solidity-by-example.rst

```diff
@@ -258,13 +263,29 @@ activate themselves.
                 throw;
             }
             if (highestBidder != 0) {
-                highestBidder.send(highestBid);
+                // Sending back the money by simply using
+                // highestBidder.send(highestBid) is a security risk
+                // because it can be prevented by the caller by e.g.
+                // raising the call stack to 1023. It is always safer
+                // to let the recipient withdraw their money themselves. 
+                pendingReturns[highestBidder] += highestBid;
             }
             highestBidder = msg.sender;
             highestBid = msg.value;
             HighestBidIncreased(msg.sender, msg.value);
         }
 
+        /// Withdraw a bid that was overbid.
+        function withdraw() {
+            var amount = pendingReturns[msg.sender];
+            // It is important to set this to zero because the recipient
+            // can call this function again as part of the receiving call
+            // before `send` returns.
+            pendingReturns[msg.sender] = 0;
+            if (!msg.sender.send(amount))
+                throw; // If anything fails, this will revert the changes above
```

Comment:
I'm cautious about this throw. As it is being used now, it's fine, but if someone were to instrument a withdraw that were automated, or part of a bookkeeping function, then it's possible the contract could be wedged by a malicious ethereum wallet.